core/services/kraken/api/v1/routers/extension.py

```diff
@@ -17,48 +18,42 @@
 
 
 @extension_router_v1.post("/install", status_code=status.HTTP_201_CREATED)
-async def install_extension(extension: Extension) -> Any:
-    if not extension.is_valid():
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail="Invalid extension description",
-        )
-    if not kraken_instance.has_enough_disk_space():
-        raise HTTPException(
-            status_code=status.HTTP_507_INSUFFICIENT_STORAGE,
-            detail="Not enough disk space to install the extension",
-        )
-    compatible_digest = await kraken_instance.is_compatible_extension(extension.identifier, extension.tag)
-    # Throw an exception only if compatible_digest is False, indicating the extension is in the manifest but it is
-    # incompatible. If compatible_digest is None, we are going to trusty that the image is compatible and will work
-    if compatible_digest is False:
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail="Extension is not compatible with the current machine running BlueOS.",
-        )
-    return StreamingResponse(streamer(kraken_instance.install_extension(extension, compatible_digest)))
+async def install_extension(body: ExtensionSource) -> StreamingResponse:
+    extension = Extension(body)
+
+    return StreamingResponse(streamer(extension.install()))
 
 
 @extension_router_v1.post("/uninstall", status_code=status.HTTP_200_OK)
 async def uninstall_extension(extension_identifier: str) -> Any:
-    return await kraken_instance.uninstall_extension_from_identifier(extension_identifier)
+    extensions = cast(List[Extension], await Extension.from_settings(extension_identifier))
+
+    await asyncio.gather(*[ext.uninstall() for ext in extensions])
 
 
 @extension_router_v1.post("/update_to_version", status_code=status.HTTP_201_CREATED)
-async def update_extension(extension_identifier: str, new_version: str) -> Any:
-    return StreamingResponse(streamer(kraken_instance.update_extension_to_version(extension_identifier, new_version)))
+async def update_extension(extension_identifier: str, new_version: str) -> StreamingResponse:
+    extension = cast(Extension, await Extension.from_manifest(extension_identifier, new_version))
+
```

Comment:
just a minor thing, is this spaces necessary ? in all functions here ?

Reply:
They aren't necessary, already removed!